pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -48,13 +49,35 @@ func (c *Cluster) getLabels(monConfig *monConfig, canary, includeNewLabels bool)
 	if canary {
 		labels["mon_canary"] = "true"
 	}
-	if c.spec.Mon.VolumeClaimTemplate != nil && includeNewLabels {
-		labels["pvc_name"] = monConfig.ResourceName
+	if includeNewLabels {
+		if c.monVolumeClaimTemplate(monConfig) != nil {
+			labels["pvc_name"] = monConfig.ResourceName
+		}
+		if monConfig.Zone != "" {
+			labels["stretch-zone"] = monConfig.Zone
+		}
 	}
 
 	return labels
 }
 
+func (c *Cluster) stretchFailureDomainName() string {
+	label := c.stretchFailureDomainLabel()
+	index := strings.Index(label, "/")
+	if index == -1 {
+		return label
+	}
+	return label[index+1:]
+}
+
+func (c *Cluster) stretchFailureDomainLabel() string {
+	if c.spec.IsStretchCluster() && c.spec.Mon.StretchCluster.FailureDomainLabel != "" {
```

Comment:
```suggestion
	if c.spec.Mon.StretchCluster.FailureDomainLabel != "" {
```

Do we call this if not `c.spec.IsStretchCluster()`?

Reply:
you're right it's redundant